src/test/java/net/raisetechPractice/UserManagementApp/service/UserServiceImplTest.java

```diff
@@ -0,0 +1,106 @@
+package net.raisetechPractice.UserManagementApp.service;
+
+import net.raisetechPractice.UserManagementApp.entity.User;
+import net.raisetechPractice.UserManagementApp.form.UserForm;
+import net.raisetechPractice.UserManagementApp.repository.UserMapper;
+import org.apache.ibatis.javassist.NotFoundException;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.List;
+import java.util.Optional;
+
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class UserServiceImplTest {
+    @InjectMocks
+    UserServiceImpl userServiceImpl;
+
+    @Mock
+    UserMapper userMapper;
+
+    @Test
+    public void Userが全件取得できること() throws Exception {
+        when(userMapper.findUsers(null)).thenReturn(List.of(new User(1, "Taro", 30), new User(2, "Jiro", 25)));
+        List<User> actual = userServiceImpl.findUsers(null);
+        assertThat(actual).hasSize(2);
+    }
+
+    @Test
+    public void Userが登録されていないとき空のリストが返ること() {
+        when(userMapper.findUsers(null)).thenReturn(List.of());
+        List<User> actual = userServiceImpl.findUsers(null);
+        assertThat(actual).hasSize(0);
+    }
+
+    @Test
+    public void 該当する年齢のUserが全件取得できること() throws Exception {
+        when(userMapper.findUsers(25)).thenReturn(List.of(new User(2, "Jiro", 25), new User(4, "Kazuko", 25)));
+        List<User> actual = userServiceImpl.findUsers(25);
+        int age = actual.get(0).getAge();
+        assertThat(actual).hasSize(2);
+        assertThat(age).isEqualTo(25);
+    }
+
+    @Test
+    public void 該当する年齢のUserが登録されていないとき空のリストが返ること() {
+        when(userMapper.findUsers(35)).thenReturn(List.of());
+        List<User> actual = userServiceImpl.findUsers(35);
+        assertThat(actual).hasSize(0);
+    }
+
+    @Test
+    public void 該当するIDのUserインスタンスが取得できること() throws Exception {
+        when(userMapper.findUserById(1)).thenReturn(Optional.of(new User(6, "Yoko", 20)));
+        int actual = userServiceImpl.findUserById(1).getId();
+        assertThat(actual).isEqualTo(6);
+    }
+
+    @Test
+    public void 検索時に該当するIDのUserがいないときエラーが発生すること() {
+        when(userMapper.findUserById(1)).thenReturn(Optional.ofNullable(null));
```

Comment:
`Optional.empty()`を使うのがより適切ですね。
```suggestion
        when(userMapper.findUserById(1)).thenReturn(Optional.empty());
```

Reply:
`Optional.empty()`を使うように修正しました。
https://github.com/soishiki/Lesson13/pull/2/commits/b9e6f4fc9d9dd5a0903eed807ab3f476f2a62871